### PR TITLE
MQTT code optimization/formatting

### PR DESCRIPTION
**Pull request by Nerivec**
- Fix missing await

@Koenkk can you take a look at the `onConnect` function. Seems like the variable assignment in `connect` and the `@bind` are unnecessary? Maybe the whole function's body can just be moved in the `connect` event?

### CODE REVIEW

**Comment by Koenkk** [comment]
I think the var assignment is required (failing test should provide proof of that when removing), but the bind isnt

File: lib/mqtt.ts [hunk 1]
```diff
@@ -78,14 +79,14 @@ export default class MQTT {
             this.client = mqtt.connect(mqttSettings.server, options);
             // @ts-ignore https://github.com/Koenkk/zigbee2mqtt/issues/9822
             this.client.stream.setMaxListeners(0);
-            this.eventBus.onPublishAvailability(this, () => this.publishStateOnline());
+            this.eventBus.onPublishAvailability(this, this.publishStateOnline.bind(this));
```
Comment (Koenkk) [inline lib/mqtt.ts:82]: Lets add the @bind that instead of .bind(this)

Comment (Nerivec) [reply lib/mqtt.ts]: Old habits! 😅

**Comment by Nerivec** [comment]
I'm tempted to move `onConnect()` in-place instead of having a function call inside an anon function, not much difference in readability, and everything would be... in-place, but I'll let you decide.

**Comment by Koenkk** [comment]
Thats also fine!

**Pull request merged**